index.bs

```diff
@@ -10651,6 +10657,155 @@ The [=remote end steps=] given |session| and |command parameters| are:
 
 </div>
 
+## The autofill Module ## {#module-autofill}
+
+The <dfn export for=modules>autofill</dfn> module contains functionality for
+saving and triggering autofill.
+
+A [=/browsing context=] has a <dfn>autofill store for testing</dfn>, an
+<a>ordered map</a>.
+
+### Definition ### {#module-autofill-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+
+InputCommand = (
+  autofill.save //
+  autofill.trigger
+)
+</pre>
+
+### Commands ### {#module-autofill-commands}
+
+#### The autofill.save Command ####  {#command-autofill-save}
+
+The <dfn export for=commands>autofill.save</dfn> command saves a set of field
+name values to the [=/browsing context=]'s [=autofill store for testing=].
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      autofill.save = (
+        method: "autofill.save",
+        params: autofill.saveParameters
+      )
+
+      autofill.saveParameters = {
+        context: browsingContext.BrowsingContext,
+        fields: [*autofill.field]
```

Comment:
@OrKoN trying to make sure I fully understand the concern here:

> what I worry about is that the users would need to know which set of attributes actually can be successfully saved in a given implementation

I'm nor super familiar with the descriptive language of webdriver-bidi in this file, is there any way to define an enumeration that specifies which keys are allowed? These keys would 1:1 map to the keys in the autofill specification, and hence would kind of mimic the form-based approach, just without specifying the form. Kind of a "shortcut".

>what to do if the fields from of incompatible items are being sent at once such as fields from an address and from a credit card

I'm trying to understand where this would cause problems? If the setup state of autofill for the test is to have credit card and address information stored, and then the test only creates forms for addresses and only asserts on addresses and passes, wouldn't this be good and actually real world conditions, as I'm sure I have way more autofill entries stored than a page renders.

>  it seems like saving step can be left out and the fields can be sent directly to trigger. An explicit save step would also need some way to clean up the data (e.g., between tests).

I think we are on the same page re. the explicit `save` vs passing in state for every test, as I suggested [here](https://github.com/w3c/webdriver-bidi/pull/706/files#r1615990960). I believe our 2 proposals are essentially the same, right?

Reply:
@martinLechnerShopify now pushed the parameters onto `trigger`. @OrKoN - PTAL?